src/utils/Fetcher.ts

```diff
@@ -49,16 +73,30 @@ export class Fetcher {
     variables: Record<string, any>,
     options?: RequestOptions
   ): Promise<T> {
-    return queryGraph(url, query, variables, mergeRequestOptions(this.customDefaults, options))
+    return queryGraph(url, query, variables, mergeRequestOptions(this.customDefaults, options), this.middleware)
   }
 }
 
-export async function fetchJson(url: string, options?: RequestOptions): Promise<any> {
-  return fetchInternal(url, response => response.json(), mergeRequestOptions(FETCH_JSON_DEFAULTS, options))
+export async function fetchJson(url: string, options?: RequestOptions, middleware?: FetcherMiddleware): Promise<any> {
```

Comment:
Would it make sense to add the middleware to the `RequestOptions`? I worry that we usually end up merging this parameters and therefore breaking the interface